app/assets/javascripts/ui_models/app_state.ts

```diff
@@ -148,6 +148,62 @@ class NoAccountWarningState {
   }
 }
 
+class SearchOptions {
```

Comment:
So many classes in this file. What's your take on this?

Reply:
I could go either way honestly. 
These classes are only meant to be consumed by AppState so they only exist to separate behavior in a sensible way. In that sense they're not different from one big class with many methods, it's just one holding class containing smaller objects. Extracting each class to its own separate file would the overall intent (to have one global state object) less clear I think.

Reply:
I don't believe extracting each class into its separate file would cause much harm either so I have no quarrel with doing so if you have a stronger opinion about this than I do 🙂

Reply:
Just cleaner to do so imo. Can be in a folder within a app_state dir. Similar to creating a component that has many components: I think it's cleaner if each component were its own file. Especially as the sub components begin to grow. In my experience it's also frustrating to open something like app_state and expect to see AppState but then having to scroll to who knows where to actually begin seeing the AppState interface.

Reply:
Gotcha 👍